pkg/server/chaosd/host_attack.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosd
+
+import (
+	"context"
+
+	"github.com/chaos-mesh/chaosd/pkg/core"
+	"github.com/google/uuid"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+)
+
+type HostManager interface {
+	Name() string
+	Shutdown() error
+}
+
+func (s *Server) HostAttack(attack *core.HostCommand) (uid string, err error) {
+	uid = uuid.New().String()
+
+	if err = s.exp.Set(context.Background(), &core.Experiment{
+		Uid:            uid,
+		Status:         core.Created,
+		Kind:           core.HostAttack,
+		Action:         attack.Action,
+		RecoverCommand: attack.String(),
+	}); err != nil {
+		err = errors.WithStack(err)
+		return
+	}
+
+	defer func() {
```

Comment:
@WangXiangUSTC This function that's deferred could be made a utility function since it's used a lot?

Reply:
Yes, I agree with you

Reply:
So should I include all these refactor changes here or in another PR?

Reply:
I think it's better to refactor it in another PR